cfg_mgmt/reporting.py

```diff
@@ -30,6 +30,60 @@ class CfgElementStatusReport:
     status: typing.Optional[cmm.CfgStatus]
 
 
+def analyse_cfg_element_status(
+    cfg_element_status: CfgElementStatusReport,
+) -> cmm.CfgStatusAnalysis:
+
+    analysis = cmm.CfgStatusAnalysis(
+        fullyCompliant=True,
+        hasResponsible=True,
+        hasRule=True,
+        assignedRuleRefersToUndefinedPolicy=False,
+        hasStatus=True,
+    )
+
+    if not cfg_element_status.responsible:
+        analysis.fullyCompliant = False
+        analysis.hasResponsible = False
+
+    if not cfg_element_status.rule:
+        analysis.fullyCompliant = False
+        analysis.hasRule = False
+
+    elif not cfg_element_status.policy:
+        analysis.fullyCompliant = False
+        analysis.assignedRuleRefersToUndefinedPolicy = True
+
+    else:
+        # have rule w/ policy
```

Comment:
instead of writing this comment, you might express this condition with an additional `elif` (as opposed to using an else w/ implicit conditions). also add `else: raise NotImplemented # or RuntimeError` to guard against programming errors